File: backend/src/main/java/com/examly/springapp/service/EmployeeService.java
package com.examly.springapp.service;

import com.examly.springapp.dto.EmployeeDTO;
import java.util.List;

public interface EmployeeService {
    EmployeeDTO createEmployee(EmployeeDTO dto);
    EmployeeDTO getEmployeeById(Long id);
    EmployeeDTO updateEmployee(Long id, EmployeeDTO dto);
    List<EmployeeDTO> getOrgChart();
    List<EmployeeDTO> searchEmployees(String query);
    List<EmployeeDTO> getAllEmployees();
    void deleteEmployee(Long id);
}

File: frontend/src/services/employeeService.js
import api from './api';

export const employeeService = {
  getAllEmployees: async () => {
    const response = await api.get('/api/employees');
    return response.data;
  },

  getEmployeeById: async (id) => {
    const response = await api.get(`/api/employees/${id}`);
    return response.data;
  },

  createEmployee: async (employeeData) => {
    const response = await api.post('/api/employees', employeeData);
    return response.data;
  },

  updateEmployee: async (id, employeeData) => {
    const response = await api.put(`/api/employees/${id}`, employeeData);
    return response.data;
  },

  deleteEmployee: async (id) => {
    const response = await api.delete(`/api/employees/${id}`);
    return response.data;
  },

  searchEmployees: async (query) => {
    const response = await api.get(`/api/employees/search?query=${encodeURIComponent(query)}`);
    return response.data;
  },

  getOrgChart: async () => {
    const response = await api.get('/api/employees/org-chart');
    return response.data;
  }
};

export default employeeService;
